src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useIsApprovalNeeded.tsx

```diff
@@ -0,0 +1,60 @@
+import type { AccountId } from '@shapeshiftoss/caip'
+import { useMemo } from 'react'
+import { useAllowance } from 'components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useAllowance'
+import { bn } from 'lib/bignumber/bignumber'
+import type { TradeQuoteStep } from 'lib/swapper/types'
+import { assertUnreachable } from 'lib/utils'
+
+import { GetAllowanceErr } from '../../../hooks/useAllowanceApproval/helpers'
+
+export type UseIsApprovalNeededProps = {
+  watch: boolean
+  tradeQuoteStep: TradeQuoteStep | undefined
+  sellAssetAccountId: AccountId | undefined
+}
+
+export const useIsApprovalNeeded = ({
+  watch,
+  tradeQuoteStep,
+  sellAssetAccountId,
+}: UseIsApprovalNeededProps) => {
+  const { isLoading, data } = useAllowance({
```

Comment:
This won't do what it may seem like it doe and will only work on the first fetch, but won't work for refetches, where we should check `isRefetching`.
Note, if tackling https://github.com/shapeshift/web/pull/5774/files#r1418024728, then that becomes fine as we won't be *re*fetching

Reply:
90b8d200ec3a4c490d0be04ffbb0276c4ada23d2